File: TelerikUIExamplesInSwift2.0/TelerikUIExamplesInSwift/DefaultAnimation.swift
//
//  DefaultAnimation.swift
//  Swift Examples
//
//  Copyright (c) 2014 Telerik. All rights reserved.
//

import Foundation
import QuartzCore

class DefaultAnimation: ExampleViewController {
    
    let chart = TKChart()
    
    override init(nibName nibNameOrNil: String?, bundle nibBundleOrNil: NSBundle?) {
        super.init(nibName: nibNameOrNil, bundle: nibBundleOrNil)

        let names = ["Area Series", "Pie Series", "Line Series", "Scatter Series", "Bar Series", "Column Series"]
        let idiom = UIDevice.currentDevice().userInterfaceIdiom
        
        self.addOption(self.nameForOption(0, names: names, idiom: idiom)) { self.setupAreaSeries() }
        self.addOption(self.nameForOption(1, names: names, idiom: idiom)) { self.setupPieSeries() }
        self.addOption(self.nameForOption(2, names: names, idiom: idiom)) { self.setupLineSeries() }
        self.addOption(self.nameForOption(3, names: names, idiom: idiom)) { self.setupScatterSeries() }
        self.addOption(self.nameForOption(4, names: names, idiom: idiom)) { self.setupBarSeries() }
        self.addOption(self.nameForOption(5, names: names, idiom: idiom)) { self.setupColumnSeries() }
        
        self.selectedOption = 2
    }

    required init(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        chart.frame = self.exampleBoundsWithInset
        chart.autoresizingMask = UIViewAutoresizing(rawValue:UIViewAutoresizing.FlexibleWidth.rawValue | UIViewAutoresizing.FlexibleHeight.rawValue)
        chart.allowAnimations = true
        self.view.addSubview(chart)
        
        self.setupLineSeries()
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    func nameForOption(index: NSInteger, names: NSArray, idiom: UIUserInterfaceIdiom) -> String {
        var name = names[index] as! String
        if (idiom != UIUserInterfaceIdiom.Pad) {
            name = "\(name) Animation"
        }
        return name
    }
    
    func setupLineSeries() {
        chart.removeAllData()
        
        var array = [TKChartDataPoint]()
        for i in 0..<10 {
            array.append(TKChartDataPoint(x: i, y: Int(arc4random() % 100)))
        }
        
        let lineSeries = TKChartLineSeries(items: array)
        lineSeries.selectionMode = TKChartSeriesSelectionMode.Series
        chart.addSeries(lineSeries)
    }
    
    func setupAreaSeries() {
        chart.removeAllData()
        
        var array = [TKChartDataPoint]()
        for i in 0..<10 {
            array.append(TKChartDataPoint(x: i, y: Int(arc4random() % 100)))
        }
        
        let areaSeries = TKChartAreaSeries(items: array)
        areaSeries.selectionMode = TKChartSeriesSelectionMode.Series
        chart.addSeries(areaSeries)
    }
    
    func setupPieSeries() {
        chart.removeAllData()
        
        var array = [TKChartDataPoint]()
        array.append(TKChartDataPoint(name: "Google", value: 20))
        array.append(TKChartDataPoint(name: "Apple", value: 30))
        array.append(TKChartDataPoint(name: "Microsoft", value: 10))
        array.append(TKChartDataPoint(name: "IBM", value: 5))
        array.append(TKChartDataPoint(name: "Oracle", value: 8))
        
        let series = TKChartPieSeries(items: array)
        series.selectionMode = TKChartSeriesSelectionMode.DataPoint
        chart.addSeries(series)
    }
    
    func setupScatterSeries() {
        chart.removeAllData()
        
        var points = [TKChartDataPoint]()
        for _ in 0..<100 {
            points.append(TKChartDataPoint(x: Int(arc4random() % 1450), y: Int(arc4random() % 150)))
        }
        
        let scatterSeries = TKChartScatterSeries(items: points)
        scatterSeries.selectionMode = TKChartSeriesSelectionMode.Series
        chart.addSeries(scatterSeries)
    }
    
    func setupBarSeries() {
        chart.removeAllData()
        
        var array = [TKChartDataPoint]()
        for i in 0..<10 {
            array.append(TKChartDataPoint(x: Int(arc4random() % 100), y: i))
        }
        
        let barSeries = TKChartBarSeries(items: array)
        barSeries.selectionMode = TKChartSeriesSelectionMode.Series
        chart.addSeries(barSeries)
    }
    
    func setupColumnSeries() {
        chart.removeAllData()
        
        var array = [TKChartDataPoint]()
        for i in 0..<10 {
            array.append(TKChartDataPoint(x: i, y: Int(arc4random() % 100)))
        }
        
        let columnSeries = TKChartColumnSeries(items: array)
        columnSeries.selectionMode = TKChartSeriesSelectionMode.Series
        chart.addSeries(columnSeries)
    }
}

File: Examples/Examples/Chart/Animations/DefaultAnimation.cs
using System;
using System.Collections.Generic;

using Foundation;
using UIKit;

using TelerikUI;

namespace Examples
{
	[Register("DefaultAnimation")]
	public class DefaultAnimation: XamarinExampleViewController
	{
		TKChart chart;

		public override void ViewDidLoad ()
		{
			this.AddOption ("Area Series", setupAreaSeries);
			this.AddOption ("Pie Series", setupPieChart);
			this.AddOption ("Line Series", setupLineSeries);
			this.AddOption ("Scatter Series", setupScatterSeries);
			this.AddOption ("Bar Series", setupBarSeries);
			this.AddOption ("Column Series", setupColumnSeries);

			base.ViewDidLoad ();

			chart = new TKChart (this.View.Bounds);
			chart.AutoresizingMask = UIViewAutoresizing.FlexibleWidth | UIViewAutoresizing.FlexibleHeight;
			chart.AllowAnimations = true;
			this.View.AddSubview (chart);

			this.setupAreaSeries ();
		}
			
		public void setupLineSeries()
		{
			chart.RemoveAllData ();

			Random r = new Random ();
			List<TKChartDataPoint> list = new List<TKChartDataPoint> ();
			for (int i = 0; i < 10; i++) {
				list.Add (new TKChartDataPoint(new NSNumber(i), new NSNumber(r.Next() % 100)));
			}

			chart.AddSeries (new TKChartLineSeries (list.ToArray()));
		}

		public void setupAreaSeries()
		{
			chart.RemoveAllData ();

			Random r = new Random ();
			List<TKChartDataPoint> list = new List<TKChartDataPoint> ();
			for (int i = 0; i < 10; i++) {
				list.Add (new TKChartDataPoint(new NSNumber(i), new NSNumber(r.Next() % 100)));
			}

			TKChartAreaSeries areaSeries = new TKChartAreaSeries (list.ToArray());
			areaSeries.SelectionMode = TKChartSeriesSelectionMode.Series;
			chart.AddSeries (areaSeries);
		}

		public void setupPieChart()
		{
			chart.RemoveAllData ();

			List<TKChartDataPoint> list = new List<TKChartDataPoint> ();
			list.Add (new TKChartDataPoint(new NSNumber(20), null, "Google"));
			list.Add (new TKChartDataPoint(new NSNumber(30), null, "Apple"));
			list.Add (new TKChartDataPoint(new NSNumber(10), null, "Microsoft"));
			list.Add (new TKChartDataPoint(new NSNumber(5), null, "IBM"));
			list.Add (new TKChartDataPoint(new NSNumber(8), null, "Oracle"));
			TKChartPieSeries series = new TKChartPieSeries (list.ToArray());
			series.SelectionMode = TKChartSeriesSelectionMode.DataPoint;
			chart.AddSeries (series);
		}

		public void setupScatterSeries()
		{
			chart.RemoveAllData ();

			Random r = new Random ();
			List<TKChartDataPoint> list = new List<TKChartDataPoint> ();
			for (int i = 0; i < 100; i++) {
				list.Add (new TKChartDataPoint(new NSNumber(r.Next() % 1450), new NSNumber(r.Next() % 150)));
			}

			TKChartScatterSeries scatterSeries = new TKChartScatterSeries (list.ToArray());
			scatterSeries.SelectionMode = TKChartSeriesSelectionMode.Series;
			chart.AddSeries (scatterSeries);
		}

		public void setupBarSeries()
		{
			chart.RemoveAllData ();

			Random r = new Random ();
			List<TKChartDataPoint> list = new List<TKChartDataPoint> ();
			for (int i = 0; i < 10; i++) {
				list.Add (new TKChartDataPoint(new NSNumber(r.Next() % 100), new NSNumber(i)));
			}

			TKChartBarSeries barSeries = new TKChartBarSeries (list.ToArray());
			barSeries.SelectionMode = TKChartSeriesSelectionMode.Series;
			chart.AddSeries (barSeries);
		}

		public void setupColumnSeries()
		{
			chart.RemoveAllData ();

			Random r = new Random ();
			List<TKChartDataPoint> list = new List<TKChartDataPoint> ();
			for (int i = 0; i < 10; i++) {
				list.Add (new TKChartDataPoint(new NSNumber(i), new NSNumber(r.Next() % 100)));
			}

			TKChartColumnSeries columnSeries = new TKChartColumnSeries (list.ToArray());
			columnSeries.SelectionMode = TKChartSeriesSelectionMode.Series;
			chart.AddSeries (columnSeries);
		}
	}
}

